client/src/components/RelatedItems/Outfits.jsx

```diff
@@ -1,175 +1,171 @@
-        total += +(entry?.[0] ?? 0) * +(entry?.[1] ?? 0);
-      });
-      let averageRate = total / count;
-    
+
       const newSlideContent = {
         id: productId,
-        ...goodsDetail,
-        averageRate,
+        ...detailResponse.data.results[0],
+        averageRate: calculateAverageRate(reviewsMetaResponse?.data?.ratings),
       };
-    
-      const updatedSlides = [...slides, newSlideContent];
-      setSlides(updatedSlides);
-      setRatings({...ratings, [productId]: averageRate});
-    
-      localStorage.setItem('slide', JSON.stringify(updatedSlides));
+
+      setSlides(prevSlides => [...prevSlides, newSlideContent]);
+      localStorage.setItem('slide', JSON.stringify([...slides, newSlideContent]));
     } catch (error) {
       console.error('Error adding outfit: ', error);
     }
   };
 
-  const handleRatingChange = (id, newRating) => {
-    setRatings({...ratings, [id]: newRating});
+  const calculateAverageRate = (ratings) => {
+    let total = 0, count = 0;
+    for (const [rate, number] of Object.entries(ratings || {})) {
+      total += rate * number;
+      count += number;
+    }
+    return count > 0 ? total / count : 0;
+  };
+
+  const handleNext = () => {
+    setCurrentSlide((prevSlide) => (prevSlide + 1) % slides.length);
   };
+
+  const handlePrev = () => {
+    setCurrentSlide((prevSlide) => prevSlide === 0 ? slides.length - 1 : prevSlide - 1);
+  };
+
+  // 样式定义，保持与 RelatedProducts 组件一致
+  const carouselContainerStyle = {
+    position: 'relative',
+    width: '100%',
+    overflow: 'hidden',
+  };
+  
+  const carouselWrapperStyle = {
+    display: 'flex',
+    transform: `translateX(-${currentSlide * 30}%)`, 
+    transition: 'transform 0.3s ease-in-out',
+  };
+  
+  const carouselSlideStyle = {
+    flex: '0 0 30%',
+    opacity: 1,
+    transition: 'opacity 0.3s ease-in-out',
+    margin: '0 5px',
+  };
+
+  const imageStyle = {
+    height: '250px',
+    backgroundColor: 'grey',
+    position: 'relative',
+    overflow: 'hidden',
+    backgroundSize: 'cover',
+    backgroundRepeat: 'no-repeat',
+  };
+  const slideImageStyle = {
+    height: '380px',
+    width: '250px',
+    border: 'solid',
+    display: 'flex',
+    flexDirection: 'column',
+    position: 'relative', 
+    }
+  const firststyle = {
+    height: '380px',
+    width: '250px',
+    border: 'solid',
+    display: 'flex',
+    flexDirection: 'column',
+    position: 'relative', 
+    backgroundColor: '#e9e9e9',
+    justifyContent: 'center',
+    alignItems: 'center',
+  }
+
+  const carouselControlStyle = {
+    position: 'absolute',
+    top: '50%',
+    transform: 'translateY(-50%)',
+    backgroundColor: '#fff',
+    border: 'none',
+    cursor: 'pointer',
+  };
+
+  const carouselPrevStyle = {
+    ...carouselControlStyle,
+    left: '10px',
+  };
+
+  const carouselNextStyle = {
+    ...carouselControlStyle,
+    right: '10px',
+  };
+  const OutfitStyles = {
+    maxWidth: '80%', 
+    margin: '0 auto', 
```

Comment:
Change this to `margin: '0 6rem'` and remove the maxWidth property to conform with the rest of the page.